src/main/java/org/roaringbitmap/longlong/RoaringTreeMap.java

```diff
@@ -0,0 +1,284 @@
+package org.roaringbitmap.longlong;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.roaringbitmap.IntIterator;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+//  this class is not thread-safe
+//@Beta
+public class RoaringTreeMap {
+	protected final NavigableMap<Integer, MutableRoaringBitmap> hiToBitmap = new TreeMap<>();
+
+	// Prevent recomputing all cardinalities when requesting consecutive ranks
+	protected boolean allValid = false;
+	protected int firstHiNotValid = Integer.MIN_VALUE;
+
+	// TODO: I would prefer not managing arrays myself
+	protected long[] sortedCumulatedCardinality = new long[0];
+	protected int[] sortedHighs = new int[0];
+
+	protected final List<MutableRoaringBitmap> linkedBitmaps = new ArrayList<>();
+
+	// Prevent indirection when writing consecutive Integers
+	private transient Map.Entry<Integer, MutableRoaringBitmap> latest = null;
```

Comment:
Ok. Then, you should make all performance attributes transient as well, as they duplicate information. I am thinking about allValid, firstHiNotValid, sorted* and linkedBitmaps. LinkedBitmaps and sortedHighs can be rebuilt during deserialization, the others are already handled by a lazy mechanism to be populated on demand.